### ISSUE TITLE
`python setup.py check --restructuredtext` misses an error with python3.5

### ISSUE THREAD

**Issue by paternal** [pypa/readme_renderer#57]
With commit 2d4fd7f7e9e768d4f09b04e463de3270323a0c2e of [this repo](https://framagit.org/spalax/papersize), I get:

- python2.7, python3.6:

  ~~~sh
  $ python setup.py check --restructuredtext
  running check
  warning: Check: :2: (WARNING/2) Title underline too short.
  
  warning: Check:
  
  warning: Check: papersize 📏 Paper size related tools
  
  warning: Check: ====================================
  
  warning: Check: Invalid markup which will not be rendered on PyPI.
  ~~~

- python3.5:

  ~~~sh
  $ python setup.py check --restructuredtext
  running check
  ~~~

In short: python3.5 does not see an error that is correctly raised with python2.7 and python3.6 (I did not test any other python version).

-- Louis

**Comment by di** [comment]
The issue seems to be that docutils is interpreting the length of the emoji character incorrectly.

I've filed a corresponding report here: https://sourceforge.net/p/docutils/bugs/335/

**Comment by di** [comment]
Docutils determines the length of these characters based on the stdlib `unicodedata.east_asian_width` function. This is fixed upstream in 2.7 and 3.6 and since the impact is minimal, the maintainers aren't interested in adding a workaround for earlier versions of Python.

I don't think it makes sense to add such a workaround in `readme_renderer` either, so I'm going to close this as a known issue that we won't fix, but thanks for reporting!
